src/runtime_src/core/tools/xbutil2/SubCmdJsonObjects.h

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022-2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <string>
+#include <variant>
+#include <unordered_map>
+
+#include "boost/property_tree/ptree.hpp"
+#include "boost/program_options.hpp"
+
+static constexpr std::string_view const_name_literal = "name";
+static constexpr std::string_view const_description_literal = "description";
+static constexpr std::string_view const_tag_literal = "tag";
+static constexpr std::string_view const_alias_literal = "alias";
+static constexpr std::string_view const_default_value_literal = "default_value";
+static constexpr std::string_view const_option_type_literal = "option_type";
+static constexpr std::string_view const_value_type_literal = "value_type";
+static constexpr std::string_view const_options_literal = "options";
+
+namespace po = boost::program_options;
+namespace pt = boost::property_tree;
+
+using variantType = std::variant<bool, std::string, std::vector<std::string>>;
+
+enum class ValueType {
+  BOOL,
+  STRING,
+  ARRAY,
+  NONE
+};
```

Comment:
Never use UPPERCASE for class enumerators; these are bound to be replaced by some stupid braindead macro from some place.  Class enumerators are scoped, so keep them lower case and short like you have them here.